.github/workflows/application-signals-python-e2e-ec2-test.yml

```diff
@@ -0,0 +1,160 @@
+## Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+## SPDX-License-Identifier: Apache-2.0
+
+# This is a reusable workflow for running the Python E2E Canary test for Application Signals.
+# It is meant to be called from another workflow.
+# Read more about reusable workflows: https://docs.github.com/en/actions/using-workflows/reusing-workflows#overview
+name: Application Signals Enablement E2E Testing - Python EC2 Use Case
+on:
+  workflow_call:
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  AWS_DEFAULT_REGION: us-east-1
+  APP_SIGNALS_E2E_TEST_ACCOUNT_ID: ${{ secrets.APP_SIGNALS_E2E_TEST_ACCOUNT_ID }}
+  SAMPLE_APP_ZIP: s3://aws-appsignals-sample-app-prod-us-east-1/python-sample-app.zip
+  METRIC_NAMESPACE: ApplicationSignals
+  LOG_GROUP_NAME: /aws/application-signals/data
+  TEST_RESOURCES_FOLDER: ${GITHUB_WORKSPACE}
+  GET_CW_AGENT_RPM_COMMAND: "aws s3 cp s3://${{ secrets.S3_INTEGRATION_BUCKET }}/integration-test/binary/${{ github.sha }}/amazon_linux/amd64/latest/amazon-cloudwatch-agent.rpm ./cw-agent.rpm"
+  GET_ADOT_WHEEL_COMMAND: "aws s3 cp s3://metric-schema-changes/aws_opentelemetry_distro-0.2.0-py3-none-any.whl ./aws_opentelemetry_distro-0.2.0-py3-none-any.whl && python3.9 -m pip install aws_opentelemetry_distro-0.2.0-py3-none-any.whl"
```

Comment:
Double checking if this is correct

Reply:
GET_CW_AGENT_RPM_COMMAND: This is where the bucket the cw-agent artifact built by the run is stored
GET_ADOT_WHEEL_COMMAND: This is latest image built by @thp

Reply:
```suggestion
  GET_ADOT_WHEEL_COMMAND: "aws s3 cp s3://adot-main-build-staging-jar/aws_opentelemetry_distro-0.2.0-0408d938-py3-none-any.whl ./aws_opentelemetry_distro-0.2.0-0408d938-py3-none-any.whl && python3.9 -m pip install aws_opentelemetry_distro-0.2.0-0408d938-py3-none-any.whl"
```

Reply:
Use the one that is created by the release workflow: https://github.com/aws-observability/aws-otel-python-instrumentation/actions/runs/9120281390/job/25108995603#step:11:23

Reply:
@thpierce confirmed online that this WHL is an acceptable replacement. Resolving the comment